src/components/main-view/main-view.jsx

```diff
@@ -1,77 +1,159 @@
+      .catch(error => {
+        console.error('Signup error:', error);
+        alert('Signup failed.');
+      });
   };
 
   const handleLogout = () => {
     setUser(null);
     localStorage.removeItem('token');
+    localStorage.removeItem('user');
+  };
+
+  const onFavorite = (movieId) => {
+    // Add the movie to the user's favorites
+    const updatedUser = {
+      ...user,
+      favoriteMovies: [...user.favoriteMovies, movieId]
+    };
+    setUser(updatedUser);
+    localStorage.setItem('user', JSON.stringify(updatedUser));
+
+    // You would also typically make a request to update the user's favorites on the server
+    axios.post(`https://glacial-retreat-35130-2f56298b8e37.herokuapp.com/users/${user.username}/movies/${movieId}`, {}, {
+      headers: { Authorization: `Bearer ${token}` }
+    })
+    .catch(error => console.error('Error adding favorite:', error));
   };
 
-  if (!user) {
-    return (
+  const onRemoveFavorite = (movieId) => {
+    // Remove the movie from the user's favorites
+    const updatedUser = {
+      ...user,
+      favoriteMovies: user.favoriteMovies.filter(id => id !== movieId)
+    };
+    setUser(updatedUser);
+    localStorage.setItem('user', JSON.stringify(updatedUser));
+
+    // You would also typically make a request to update the user's favorites on the server
+    axios.delete(`https://glacial-retreat-35130-2f56298b8e37.herokuapp.com/users/${user.username}/movies/${movieId}`, {
+      headers: { Authorization: `Bearer ${token}` }
+    })
+    .catch(error => console.error('Error removing favorite:', error));
+  };
+
+  return (
+    <Router>
+      <NavigationBar user={user} onLoggedOut={handleLogout}  />
       <Container>
-        <Row className="justify-content-md-center">
-          <Col md={8}>
-            {isSignup ? (
-              <SignupView onSignup={handleSignup} onSwitchToLogin={() => setIsSignup(false)} />
-            ) : (
-              <LoginView onLogin={handleLogin} onSwitchToSignup={() => setIsSignup(true)} />
-            )}
-          </Col>
+        <Row>
+          <Routes>
+            <Route path="/" element={
+              user ? (
+                movies.length > 0 ? (
+                  <Row>
+                    {movies.map(movie => (
+                      <Col md={4} key={movie._id}>
+                        <MovieCard
+                          movie={movie}
+                          onFavorite={onFavorite}
+                          onRemoveFavorite={onRemoveFavorite}
+                          isFavorite={user.favoriteMovies?.includes(movie._id)}
+                        />
+                      </Col>
+                    ))}
+                  </Row>
+                ) : (
+                  <p>Loading movies...</p>
+                )
+              ) : (
+                <Navigate to="/login" replace />
+              )
+            } />
+            <Route path="/login" element={<LoginView onLogin={handleLogin} />} />
```

Comment:
- Make this update to your `login` route.

![](https://coach-courses-us.s3.amazonaws.com/comments/users/40649/9b7263f8440a0552feba47943af62ac7/Screenshot-2024-08-17-094104.png)